### ISSUE TITLE
Propose installing Python on new project

### ISSUE THREAD

**Issue by huguesv** [Microsoft/PTVS#742]
With the integration of PTVS inside VS 2015 installer, we have a much greater number of developers that have PTVS installed (they just check to install everything in VS setup).  Many of them don't have a Python interpreter on their machine.

We already have a warning and a web page that presents the user with interpreter options when they hit F5 to run their project and have no interpreter installed.

However, between the time that they create the project, and the time they get to hit F5, there are several things they might do that trigger error conditions or disable functionality due to missing interpreter.

When creating a new project, we should check that the user has a compatible interpreter for that project, and popup a task dialog box that gives them the option of downloading and installing one immediately.

Having them install Python right from the beginning means they won't have to wait as long when they are ready to hit F5.

It could also be implemented as the same web page we display when they hit F5, but do it when the project opens.  We would have to make sure that it shows up as the active document.  There's potential competition with the survey/news page.

**Comment by zooba** [comment]
That "potential competition" at the end is a real concern - VS will reuse web pages and never show the other one (this is what happens when you create multiple web roles at once).

Do we need to maintain download links for all versions of Python? What happens when someone opens a project for 3.3 but they already have 3.4? What if it's just a 32-bit/64-bit conflict? What about virtual environments where the environment is there but the base interpreter is a mismatched version?

**Comment by int19h** [comment]
At the very least, when the project references an interpreter that is not installed, but we know what it is (i.e. it's a well-known GUID for auto-detected Python 3.3, for example), then we should show it as "Python 3.3 (missing)" or something along these lines in Solution Explorer, rather than "Unknown Python 3.3", and reflect that in the error message when you run it.

I don't think we need to author separate pages for all the different interpreters. We can just provide a download link in the task dialog that takes the user directly to the download page for that Python version (so we'd only need to maintain the fwlinks).

**Comment by zooba** [comment]
That sounds like it'll have to restructure how we deal with environments internally if it's ever going to scale. What do we show for Canopy's interpreter (which has its own GUID that we don't "know")? Or how do we provide an extensibility point for it?

I'm totally okay with redoing the internals here, and I think breaking the public API here is acceptable in 3.0, but there is a big jump between "hard-code checks for CPython" vs. "don't hard-code GUIDs" vs. "let the interpreter provider participate".

**Comment by huguesv** [comment]
I was mostly concerned with the new project scenario, not open existing project.  That's also an interesting (and more complex) case.  My thinking is that most people that start out and don't know they need an interpreter will probably create a new project vs. find an existing PTVS project somewhere.

If done just for new project, it could be similar to how we did a custom wizard for node.js express project template.  When it's created, we proposed to do a npm install for them via a message box.

On the idea of showing (missing) instead of Unknown in solution explorer, we could have a command when you right click on it to "Install".  If we could have a fwlink with a query parameter that includes the interpreter information (GUID?) we could keep them up to date with the latest download link.
